test/run_test.py

```diff
@@ -1377,11 +1376,7 @@ def parse_args():
         default=IS_CI
         and (
             TEST_WITH_CROSSREF
-            or TEST_WITH_ASAN
-            or (TEST_CONFIG == "distributed" and TEST_CUDA)
-            or (IS_WINDOWS and not TEST_CUDA)
-            or TEST_CONFIG == "nogpu_AVX512"
-            or TEST_CONFIG == "nogpu_NO_AVX2"
```

Comment:
Why disable td by default for all these configs?

Reply:
These are the ones that are enabled, but most of these checks are no ops, I've stated why in the pr description